pkg/ovs/openflow/ofctrl_action_test.go

```diff
@@ -0,0 +1,1749 @@
+		},
+		{
+			name: "SendToController",
+			actionFn: func(b Action) FlowBuilder {
+				return b.SendToController([]byte{0x1}, false)
+			},
+			expectedActionField: &openflow15.NXActionController2{},
+			expectedActionStr:   "controller(id=0,reason=no_match,userdata=01,max_len=128)",
+		},
+		{
+			name: "Meter",
+			actionFn: func(b Action) FlowBuilder {
+				return b.Meter(100)
+			},
+			expectedActionField: &openflow15.ActionMeter{
+				MeterId: 100,
+			},
+			expectedActionStr: "meter:100",
+		},
+		{
+			name: "GotoTable",
+			actionFn: func(b Action) FlowBuilder {
+				return b.GotoTable(tableID1)
+			},
+			expectedActionStr: "goto_table:100",
+		},
+		{
+			name: "GotoStage",
+			actionFn: func(b Action) FlowBuilder {
+				return b.GotoStage(stageID)
+			},
+			expectedActionStr: "goto_table:100",
+		},
+		{
+			name: "CT (commit, immediate zone value)",
+			actionFn: func(b Action) FlowBuilder {
+				return b.CT(true, tableID1, 101, nil).CTDone()
+			},
+			expectedActionField: &openflow15.NXActionConnTrack{
+				Flags:        1,
+				ZoneSrc:      0,
+				ZoneOfsNbits: 101,
+				RecircTable:  100,
+			},
+			expectedActionStr: "ct(commit,table=100,zone=101)",
+		},
+		{
+			name: "CT (without commit, immediate zone value)",
+			actionFn: func(b Action) FlowBuilder {
+				return b.CT(false, tableID1, 101, nil).CTDone()
+			},
+			expectedActionField: &openflow15.NXActionConnTrack{
+				Flags:        0,
+				ZoneSrc:      0,
+				ZoneOfsNbits: 101,
+				RecircTable:  100,
+			},
+			expectedActionStr: "ct(table=100,zone=101)",
+		},
+		{
+			name: "CT (commit, register zone value)",
+			actionFn: func(b Action) FlowBuilder {
+				return b.CT(true, tableID1, 0, NewRegField(1, 0, 16)).CTDone()
+			},
+			expectedActionField: &openflow15.NXActionConnTrack{
+				Flags:        1,
+				ZoneSrc:      0x10308,
+				ZoneOfsNbits: 0x10,
+				RecircTable:  100,
+			},
+			expectedActionStr: "ct(commit,table=100,zone=NXM_NX_REG1[0..16])",
+		},
+		{
+			name: "CT (without commit, register zone value)",
+			actionFn: func(b Action) FlowBuilder {
+				return b.CT(false, tableID1, 0, NewRegField(1, 0, 16)).CTDone()
+			},
+			expectedActionField: &openflow15.NXActionConnTrack{
+				Flags:        0,
+				ZoneSrc:      0x10308,
+				ZoneOfsNbits: 0x10,
+				RecircTable:  100,
+			},
+			expectedActionStr: "ct(table=100,zone=NXM_NX_REG1[0..16])",
+		},
+		{
+			name: "Learn",
+			actionFn: func(b Action) FlowBuilder {
+				return b.Learn(tableID1, 100, 1800, 1800, 100).Done()
+			},
+			expectedActionField: &openflow15.NXActionLearn{
+				IdleTimeout: 1800,
+				HardTimeout: 1800,
+				Priority:    100,
+				Cookie:      100,
+				TableID:     100,
+			},
+			expectedActionStr: "learn(table=100,idle_timeout=1800,hard_timeout=1800,priority=100,cookie=0x64)",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			flowMod := getFlowMod(t, tc.actionFn(table.BuildFlow(1).Action()).Done())
```

Comment:
Better to check the error as well, so it might be easier to figure out where goes wrong when others add new test cases but failure at `getFlowMod`.

```suggestion
			flowMod, err := getFlowMod(t, tc.actionFn(table.BuildFlow(1).Action()).Done())
			require.NoError(t, err)
```

Reply:
The error is checked in `getFlowMod`

Reply:
Ah, right, I mixed it with the one in ofctl_flow.go. Good to know that. Thanks.